pkg/metadata/cnimetadata.go

```diff
@@ -49,7 +49,7 @@ type ContainerMetadata struct {
 
 func RecordMetadata(datadir string, network string, data ContainerMetadata) error {
 	dir := filepath.Join(datadir, network)
-	if err := os.MkdirAll(dir, 0644); err != nil {
+	if err := os.MkdirAll(dir, 0755); err != nil {
```

Comment:
Does this have any impact on upgrades? (as in, if the directory was created with 0644, and then we tried to read it below.)

Reply:
No. This is a bug. I think the only reason this works is that the directory already exists with the right permission (perhaps created by the installer).